File: 백준/9단계/소수.js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n')

const M = Number(input[0])
const N = Number(input[1])
// const M = 64
// const N = 65

function isPrimeNum(num) {
    if (num < 2) {
        return false;
    }

    for (let i = 2; i < num; i++) {
        if (num % i == 0) {
            return false
        }
    }
    return true
}

let primes = []
let sum_value = 0
for (let i = M; i <= N; i++) {
    if (isPrimeNum(i) === true) {
        primes.push(i)
        sum_value += i
    }
}

if (!primes.length) {
    console.log(-1)
} else {
    console.log(sum_value)
    console.log(Math.min.apply(null, primes))
}

File: 백준/9단계/소수.py
M = int(input())
N = int(input())

# M 이상 N 이하의 자연수 중 소수인 것을 모두 골라 이들의 합, 최소값 반환


def isPrimeNum(num):
    if num < 2:
        return False
    for i in range(2, num):
        if num % i == 0:
            return False
    return True


primes = []
for i in range(M, N+1):
    if isPrimeNum(i) == True:
        primes.append(i)

if primes:
    print(sum(primes), min(primes))
else:
    print(-1)
